packages/embed/src/apple-pay/apple-pay.ts

```diff
@@ -0,0 +1,61 @@
+import { SubjectManager } from '../subjects'
+
+declare global {
+  interface Window {
+    ApplePaySession?: ApplePaySession
+  }
+}
+
+export const createApplePayController = (
+  subjectManager: Pick<
+    SubjectManager,
+    | 'appleAbortSession$'
+    | 'startAppleSession$'
+    | 'appleValidateMerchant$'
+    | 'applePayAuthorized$'
+    | 'completeMerchantValidation$'
+    | 'appleCompletePayment$'
+    | 'appleSessionError$'
+  >
+) => {
+  let session: ApplePaySession // only a single session at a time
+
+  subjectManager.startAppleSession$.subscribe((data) => {
+    try {
+      session = new ApplePaySession(3, data)
```

Comment:
Does 3 represent the version? Does this need to be dynamic?

Reply:
👀  Good spot!